### DISCUSSION TITLE
Dynamic slicing

### DISCUSSION THREAD

**Discussion by steradiant** [casadi/casadi#3146] [Q&A]
Hello,
I want to write the entries of a smaller array into a larger array depending on a parameter. A MWE looks like this:
```
import casadi.*

x1 = SX.sym('x1', 15);
x2 = SX.sym('x2', 6);
p = SX.sym('p',1);

x1(1+p:6+p) = x2;
```

However, I get the error: "Undefined function 'colon' for input arguments of type 'casadi.SX'." Is there a way to solve this task with casadi?

**Comment by jgillis** [comment]
Works only for MX and probably needs altered syntax X1(p+(1:6))

**Comment by steradiant** [comment]
Changing my code to MX would result in a massive speed loss, however, if I change the code snippet to MX
```
import casadi.*

x1 = MX.sym('x1', 15);
x2 = MX.sym('x2', 6);
p = MX.sym('p',1);

x1(p+(1:6)) = x2;
```
I get the error 
Error using casadi.MX/paren_asgn
No matching function for overload function 'MX_paren_asgn'.  Possible prototypes are:
    PAREN_ASGN(self,MX,Sparsity)
    PAREN_ASGN(self,MX,IM)
    PAREN_ASGN(self,MX,char)
    PAREN_ASGN(self,MX,IM,IM)
    PAREN_ASGN(self,MX,IM,char)
    PAREN_ASGN(self,MX,char,IM)
    PAREN_ASGN(self,MX,char,char)
  You have: MX, MX, MX

Error in indexing (line 42)
        paren_asgn(self, v, s.subs{:});

Error in ic_assignment_mwe (line 8)
x1(p+(1:6)) = x2;

**Comment by jgillis** [comment]
This looks like an omission in the Matlab interface. Spawned #3147
